app/authentication/token.py

```diff
@@ -4,17 +4,31 @@
 from fastapi.logger import logger
 from fastapi.security import OAuth2PasswordBearer
 from httpx import Response
-from ..routes import dataset_dependency
 
-from ..utils.rw_api import who_am_i
+from ..crud import datasets
+from ..models.orm.datasets import Dataset as ORMDataset
+from ..routes import dataset_dependency
 from ..settings.globals import PROTECTED_QUERY_DATASETS
+from ..utils.rw_api import who_am_i
+
+
+def get_request_token() -> str:
+    """Get request token if it exists."""
+    return str(OAuth2PasswordBearer(tokenUrl="/token"))
```

Comment:
I'm not sure this will now work the way you have restructured it.  The Oauth2PasswordBearer() object needs to be referenced as part of a Depends dependency to actually do its checking of the current request token.  As far as I know, you are just creating an instance of OAuth2PasswordBearer and then converting it to a string.  It's only when used as a dependency that it will do the checking via the __call_ method.  See https://github.com/tiangolo/fastapi/blob/e04d397e3224d3674cf73332303b956fc3d00b3d/fastapi/security/oauth2.py#L473 .  Also, you only need to create the oauth2_schema and oauth2_schema_no_auto objects once.  I would go back to using those pre-calculated objects.  I can huddle with you if that's not clear.